jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -153,10 +154,11 @@ public TimerJobInstance getTimerByName(String jobName) {
     				EjbTimerJob job = (EjbTimerJob) info;
     				
     				EjbGlobalJobHandle handle = (EjbGlobalJobHandle) job.getTimerJobInstance().getJobHandle();
-    				localCache.putIfAbsent(jobName, handle.getTimerJobInstance());
-    				if (handle.getUuid().equals(jobName)) {
-    					logger.debug("Job  {} does match timer and is going to be returned", jobName);
+    				
+    				if (handle.getUuid().equals(jobName)) {    					
     					found = handle.getTimerJobInstance();
+    					localCache.putIfAbsent(jobName, found);
+    					logger.debug("Job {} does match timer and is going to be returned {}", jobName, found);
```

Comment:
Since we have found it and put it in the cache, we can directly return it and avoid iterating over other timers, can't we?

Reply:
good point, updated